src/main/java/org/threadly/litesockets/utils/SSLProcessor.java

```diff
@@ -216,6 +216,12 @@ public ReuseableMergedByteBuffers decrypt(final MergedByteBuffers bb) {
 
   }
   
+  public void failHandshake(Throwable t) {
+    if(!handshakeFuture.isDone()) {
```

Comment:
Same as previous comment, this does not need to be checked (and if it did need to be checked this would be a possible race condition).